src/components/accessories/patientOPD/patientOPDForm/PatientOPDForm.tsx

```diff
@@ -215,25 +238,21 @@ const PatientOPDForm: FunctionComponent<TProps> = ({
               />
             </div>
             <div className="patientOpdForm__item">
-              <TextField
-                field={formik.getFieldProps("referralFrom")}
+              <CheckboxField
                 theme="regular"
+                fieldName={"referralFrom"}
+                checked={formik.values.referralFrom === "R"}
                 label={t("opd.referralfrom")}
-                isValid={isValid("referralFrom")}
-                errorText={getErrorText("referralFrom")}
-                onBlur={formik.handleBlur}
-                type="string"
+                onChange={handleCheckboxChange("referralFrom")}
               />
             </div>
             <div className="patientOpdForm__item">
-              <TextField
-                field={formik.getFieldProps("referralTo")}
+              <CheckboxField
                 theme="regular"
+                fieldName={"referralTo"}
+                checked={formik.values.referralTo === "R"}
                 label={t("opd.referralto")}
-                isValid={isValid("referralTo")}
-                errorText={getErrorText("referralTo")}
-                onBlur={formik.handleBlur}
-                type="string"
+                onChange={handleCheckboxChange("referralTo")}
               />
```

Comment:
Move the Referral From / To fields, under the Attendance Type fields and leave the date field.